common/src/main/java/com/wynntils/features/inventory/ItemFavoriteFeature.java

```diff
@@ -38,7 +39,7 @@ public class ItemFavoriteFeature extends UserFeature {
     public Set<String> favoriteItems = new HashSet<>();
 
     @TypeOverride
-    private final Type favoriteItemsType = new TypeToken<Set<String>>() {}.getType();
+    private final Type favoriteItemsType = new TypeToken<TreeSet<String>>() {}.getType();
```

Comment:
Why TreeSet (when the type above is HashSet)? Which one needs changing?

Reply:
(I know they can differ technically, but still)

Reply:
We need TreeSet for persisting to json. Gson treats sets as json arrays. If you have a SortedSet, then the elements of the array will be sorted, otherwise they will appear in an arbitrary order in the json file. 

Internally, we don't care if the set is sorted or not, so we can use HashSet just as well, it does not really matter. If you think it looks confusing, I can update the internal type as well to be TreeSet.

Reply:
This is all really not related to the data storage, but was a result of me getting annoyed that the config json output were not stable.

Reply:
Actually, this was one of the hacks that should be resolved by the storage impl. I decided to handle it properly as well.